src/test/compile-fail/issue-17651.rs

```diff
@@ -13,6 +13,6 @@
 
 fn main() {
     // FIXME (#22405): Replace `Box::new` with `box` here when/if possible.
-    (|| Box::new(*[0_usize].as_slice()))();
-    //~^ ERROR the trait `core::marker::Sized` is not implemented for the type `[usize]`
+    (|| Box::new(*[0].as_slice()))();
+    //~^ ERROR the trait `core::marker::Sized` is not implemented for the type `[i32]`
```

Comment:
the error message we are actually getting here is:
> error: the trait `core::marker::Sized` is not implemented for the type `[_]`

which i admit, raises a semi-interesting question about whether we are better off applying int-fallback to our *error message output* (as it seems like @eddyb assumed would be done here?)